### ISSUE TITLE
Intent of the newly added String Escape and Unescape Logix Nodes

### ISSUE THREAD

**Issue by Earthmark** [Neos-Metaverse/NeosPublic#3066]
### Where have you searched for this information before making this issue?

The discord and wiki, I don't see any references there. It's a new node so I'm not expecting much to exist with it.

### Which part of the system do you need information on?

The intent of this node, it is declared to work at escaping content such as json, however the node doesn't escape " or ', it also escapes plain spaces which are not need escaping in the suggested use cases and would produce a malformed string.
I'm not sure this node accomplishes it's designed goal, but I can't figure out what the designed goal is, hence this request.

It makes sense for regexes as is described in the patch notes, is there a system inside of neos that uses a regex that requires this component?

I thought mask text on TextRenderer might do this, but it looks like that does maybe security string replacement so I don't think it's applicable.

### What information do you need to know?

Is this intended to be used with an existing logix component? I saw that logix based locale things got added, maybe it's relevant there, I'm not sure!

### Do you plan to format and contribute the provided information to Wiki?

Yes

**Comment by ProbablePrime** [comment]
![image](https://user-images.githubusercontent.com/8791132/134444187-02942818-35b3-4c95-9237-a3183b952dfa.png)
It escapes strings and other json characters that are inside strings.

**Comment by Frooxius** [comment]
My particular purpose for this node was to Unescape JSON responses from Wikipedia for display in Neos, which escape a number of characters, like \', \" or unicode ones like \uXXXX. It works for this purpose well.

It wraps an existing function from .NET for this purpose. We can potentially expand it if you need more control.

**Comment by ProbablePrime** [comment]
I'll add this to my docs queue for the wiki too :)

**Comment by Earthmark** [comment]
So for unescape sweet, we can unescape string literals!

I think my confusion is over the escape then, I'm not seeing it work well for escaping json strings?
![image](https://user-images.githubusercontent.com/1251609/134444937-27612e92-1ad9-4e44-bb62-ceb193937996.png)

**Comment by ProbablePrime** [comment]
Its for escaping the contents of JSON Values, not escaping json itself. 

E.g.:
`
{
 "Cheese": escape(<prime's favorite cheese>)
}
`

and NOT:
`escape("{"Cheese":<prime's favorite cheese>})`

**Comment by Earthmark** [comment]
In the example a json document is my value, so my json string value is a sub document of the string "{"key":"sub document"}".

In this case I'd expect the output string is the same as the middle string display where the " characters are escaped, but that's it. However it ignores the " and dislikes the ' ', which is malformed in json anyways and would make more aggressive parsers fail to parse.

**Comment by ProbablePrime** [comment]
I am now also confusion: 
![image](https://user-images.githubusercontent.com/8791132/134445480-f4d6e59d-2b5e-47a7-ac01-5adcb73445c0.png)

**Comment by ProbablePrime** [comment]
![image](https://user-images.githubusercontent.com/8791132/134445697-d0d379e6-80d3-42de-ad83-8916d5df4528.png)

**Comment by Earthmark** [comment]
It doesn't seem to aggressively remove \ markers, I think the quote is being unescaped, but the space before the quote is getting escaped in the last display node.

**Comment by ProbablePrime** [comment]
Ok, as this is using Regex escaping, it doesn't actually escape quotations. 

Oddly Quotations need escaping for JSON. 

For proper JSON escaping, as we use NewtonSoft I think https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_JsonConvert_ToString_17.htm would be better.

Epsilion's with me in world and we both came to that conclusion.

**Comment by Aerizeon** [comment]
@Earthmark It's just calling [Regex.Escape](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.escape?view=net-5.0) which doesn't handle quotes at all.
![Image](http://files.aerizeon.net/sharex/2021/2021-09-22_22-26-Neos_ezc67WBR6Z.png)

**Comment by Earthmark** [comment]
Coorito, is the goal a json escape an unescape node then? Is the intent to escape and unescape json, or regex?

**Comment by Frooxius** [comment]
Yeah we can use that one or even provide multiple profiles for different purposes. The Unescape one is a lot more general, as it will unescape a wide range of characters that can appear in strings, that are compatible with C, C# or JSON strings and number of others.

**Comment by Earthmark** [comment]
Prime claimed this, so I'm assuming some work is pending so I'll leave this open?

**Comment by ProbablePrime** [comment]
![image](https://user-images.githubusercontent.com/8791132/136899075-61c5ca49-62c0-42c3-8171-6aca5a50405b.png)
Working well, will be overwriting the color of the escape handling enum that's too much like User IMO
